bridge/client/app/_services/api.service.ts

```diff
@@ -38,44 +39,45 @@ export class ApiService {
     this._baseUrl = value;
   }
 
-  public get baseUrl() {
+  public get baseUrl(): string {
     return this._baseUrl;
   }
 
   public get environmentFilter(): {[projectName: string]: {services: string[]}} {
-    const filter = JSON.parse(localStorage.getItem(this.ENVIRONMENT_FILTER_COOKIE)) || {};
+    const item = localStorage.getItem(this.ENVIRONMENT_FILTER_COOKIE);
+    const filter = item ? JSON.parse(item) : {};
     return filter instanceof Array ? {} : filter; // old format was an array
   }
 
   public set environmentFilter(filter: {[projectName: string]: {services: string[]}}) {
     localStorage.setItem(this.ENVIRONMENT_FILTER_COOKIE, JSON.stringify(filter));
   }
 
-  public getKeptnInfo(): Observable<any> {
-    let url = `${this._baseUrl}/bridgeInfo`;
+  public getKeptnInfo(): Observable<KeptnInfoResult> {
+    const url = `${this._baseUrl}/bridgeInfo`;
     return this.http
-      .get<any>(url)
+      .get<KeptnInfoResult>(url);
   }
 
-  public getKeptnVersion(): Observable<any> {
-    let url = `${this._baseUrl}/swagger-ui/swagger.yaml`;
+  public getKeptnVersion(): Observable<string> {
+    const url = `${this._baseUrl}/swagger-ui/swagger.yaml`;
     return this.http
-      .get<any>(url, { headers: new HttpHeaders({'Access-Control-Allow-Origin': '*'}) })
+      .get<string>(url, { headers: new HttpHeaders({'Access-Control-Allow-Origin': '*'}) })
```

Comment:
I know this question is not related directly to your changes but SonarCloud is complaining about `'Access-Control-Allow-Origin': '*'`. Can we somehow restrict the `*` wildcard?

Reply:
A just saw we can actually remove the whole `getKeptnVersion` function because this information is already in the fetched metadata.